docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,363 @@
+= AddressBook Level 3 - Tutorial
+:site-section: DeveloperGuide
+:toc:
+:toc-title:
+:toc-placement: preamble
+:imagesDir: ../images/remark
+:stylesDir: ../stylesheets
+:xrefstyle: full
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:warning-caption: :warning:
+:source-highlighter: highlightjs
+endif::[]
+
+
+== Adding your first command
+
+In this tutorial, we'll walk you through the implementation of a new command -- `remark`.
+
+This command allows users of the AddressBook application to add optional remarks to people in their address book and
+edit it if required. The command should have the format of `remark INDEX r/REMARK`. An example of the command is
+`remark 2 r/Likes baseball`.
+
+We'll assume that you have already set up the development environment as outlined in the Developer's Guide.
+
+=== Create a new `Command`
+
+Looking in the `logic.command` package, you will notice that each existing command have their own class.
+All the commands implement `Command` which means that they must override `execute()`.
+By convention, `Command` returns an instance of `CommandResult` upon success and throws a `CommandException` otherwise.
+This exception is handled gracefully -- the error message provided to `CommandException` is shown in the `ResultDisplay`.
+
+.UML diagram describing the relationship between RemarkCommand and Command
+image::CommandInterface.png[width=385px, height=385px]
+{empty} +
```

Comment:
Notation doesn't match what we cover in the module
![image](https://user-images.githubusercontent.com/1673303/63442283-69f64900-c465-11e9-80ec-70f0426c26e4.png)